### ISSUE TITLE
(How) Can I add dict.zero-g.net to Firefox' search engines? 

### ISSUE THREAD

**Issue by Taranchul** [haukex/de-en-dict#34]
Hi,
I'd like to add dict.zero-g.net to the search engines in Firefox, but when I'm on the website, FF doesn't offer to add it to them. How can I do that? (If we can't yet, please treat this as a feature request.)

Thanks for keeping Beolingus alive!

**Comment by haukex** [comment]
Thanks for this suggestion! I will look into how / whether it's possible to add to Firefox's search engines.

I believe @christophfriedrich is using a search keyword (see #31), I don't have the time to test right now but I think it's [this feature](https://kb.mozillazine.org/Using_keyword_searches#Creating_bookmarks_with_keywords).

The Mozilla website also [describes](https://support.mozilla.org/en-US/kb/how-search-from-address-bar) right-clicking in the search box, but that is something this app doesn't support yet, so I'll look into that too.

**Comment by christophfriedrich** [comment]
Yes, I'm using the search keyword and yes, the two links describe exactly this feature. The bookmark I use has the URL `https://dict.zero-g.net/#q=%s` and I assigned the keyword `beo` to it, so when I type e.g. `beo autumn` into my URL bar, it will open the app with the search results for "autumn". You can add this to your bookmarks manually.

Indeed the app doesn't support the easier way to add it via the right-click menu-- I guess that's because it's not an actual form that gets submitted somewhere. I assume if Firefox understands that this field is a search field, it will offer both the keyword option in the right-click menu AND to add it as a search engine?

The screenshot shows what the right-click menu looks like on wikipedia.de -- Interestingly, this doesn't work on google.com 🤔
![image](https://github.com/user-attachments/assets/406d32f3-aebf-48ad-9ede-1797c933da3e)

**Comment by haukex** [comment]
Thanks! I can now confirm the [instructions I linked to earlier](https://kb.mozillazine.org/Using_keyword_searches#Creating_bookmarks_with_keywords) work for me as well with a URL of `https://dict.zero-g.net/#q=%s`.

I'll look into the other possibilities later. I'm hoping the right click menu shouldn't be too difficult (I hope I just have to wrap the search box in a `<form>` and perhaps change its ID).

**Comment by haukex** [comment]
Now that I've implemented #39, I think that solves this issue too, ~so I think I'll close this issue unless someone objects?~ *Edit:* Actually, I'll keep this issue open for now, since I still haven't wrapped the input box in a `<form>`.

**Comment by christophfriedrich** [comment]
Coming back to this as I'm setting up a new device and sadly still couldn't add the search with a right-click. I did a quick experiment and indeed wrapping the `<input>` in `<form>...</form>` does cause Firefox to supply the option:

![Image](https://github.com/user-attachments/assets/94b5b40f-65b4-4561-a9cf-76180f60d975)

**Comment by christophfriedrich** [comment]
Correction: For it to work _correctly_ though, i.e. so that using the keyword actually brings up the search results, the `<input>` additionally needs the attribute `name="q"` as only that results in the URL being saved as https://dict.zero-g.net/?q=%s (otherwise the `q` is missing)

**Comment by haukex** [comment]
Thanks, yes, wrapping the input in a `<form>` is something I wanted to do over on the `search` branch together with implementing an advanced search, but unfortunately I haven't had the time to work on it much. If that continues to be the case, perhaps I'll just add the `<form>` tags on the main branch. (The reason I can't just slap the tags on there without additional modifications is because I need to check the interaction between the tag and the other [handlers](https://github.com/haukex/de-en-dict/blob/d2057c33612982bfeb7098e7a5d69791a3e117da/src/js/searchbox.ts#L36) already on the input.)
